### PR TITLE
v1.13 Backports 2023-09-04

### PR DESCRIPTION
**Pull request by jibi**
* [x] #27656 (@pchaigno)
 * [x] #27572 (@bimmlerd)
 * [x] #27171 (@joestringer)
 * [x] #27818 (@mhofstetter)
 * [x] #27416 (@brb)
 * [x] #27814 (@haiyuewa)
 * [x] #27359 (@brb)
 * [x] #27706 (@tklauser)
 * [x] #27170 (@joestringer)
 * [x] #27911 (@brb)

PRs skipped due to conflicts:

 * #27831 (@tklauser)
   * skipped as v1.13 has not been switched to resource[T] for CEP
 * #27738 (@brb)
   * skipped while we figure out what to do with the related cilium CLI changes
 * #27798 (@ti-mo)
   * dropped while https://github.com/cilium/cilium/pull/27602 gets backported, as #27798 depends on that

Once this PR is merged, you can update the PR labels via:
```upstream-prs
for pr in 27656 27572 27171 27818 27416 27814 27359 27706 27170 27911; do contrib/backporting/set-labels.py $pr done 1.13; done
```
or with
```
make add-labels BRANCH=v1.13 ISSUES=27656,27572,27171,27818,27416,27814,27359,27706,27170,27911
```

### CODE REVIEW

**Review by mhofstetter — APPROVED** [review]

**Comment by jibi** [comment]
> cilium-agent is failing in ci-e2e:
> 
> > 2023-09-04T14:33:42.564306864Z level=fatal msg="failed to start: daemon creation failed: failed to finalise LB initialization: Cannot retrieve eth1 link: Link not found" subsys=daemon
> 
> It's because https://github.com/cilium/cilium/blob/v1.13/contrib/scripts/kind.sh#L37 doesn't have `secondary-network` option which was introduced by #26338. Let's backport that PR to v1.13.

:+1: 

> Another thing is that CLI is at v0.15.3 :-(

should I tick renovate to bump v1.13 as well?

**Comment by brb** [comment]
> should I tick renovate to bump v1.13 as well?

Yes, please.

**Comment by jibi** [comment]
> > should I tick renovate to bump v1.13 as well?
> 
> Yes, please.

synced offline, dropped https://github.com/cilium/cilium/pull/27738 while we understand how to sort the cilium CLI situation

**Comment by jibi** [comment]
/test-backport-1.13

**Review by tklauser — APPROVED** [review]
My change looks good, thanks.

> * [Fix propagation of namespace labels to CEP labels #27831](https://github.com/cilium/cilium/pull/27831) (@tklauser)
>   
>   * skipped as v1.13 has not been switched to resource[T] for CEP

~Will backport myself to v1.13 and v1.12.~ Not applicable, will remove corresponding backport labels.

**Review by bimmlerd — APPROVED** [review]
#27572 LGTM, thanks!

**Review by qmonnet — APPROVED** [review]
* https://github.com/cilium/cilium/pull/27798

Looks good and I tested the same commit locally to make sure it addresses the bug on 1.13 as well. Thanks for dealing with the conflict!

**Comment by jibi** [comment]
/test-1.21-4.19

**Comment by jibi** [comment]
/test-1.16-4.19

**Comment by jibi** [comment]
/test-1.18-4.19

**Comment by jibi** [comment]
/test-1.23-4.19

**Comment by jibi** [comment]
/test-1.24-4.19

**Comment by jibi** [comment]
/test-1.22-4.19

**Review by pchaigno — APPROVED** [review]
My PR looks good. Thanks!

**Review by brb — APPROVED** [review]
Thanks!

**Review by joestringer — APPROVED** [review]
:ship:

File: .github/actions/cilium-config/action.yml [hunk 1]
```diff
@@ -37,6 +37,15 @@ inputs:
   host-fw:
     description: 'Enable host firewall'
     default: false
+  mutual-auth:
+    description: 'Enable mTLS-based Mutual Authentication'
+    default: true
```
Comment (joestringer) [inline .github/actions/cilium-config/action.yml]: Does this line make sense? v1.13 doesn't have mTLS auth.

Comment (jibi) [reply .github/actions/cilium-config/action.yml]: I'll remove it :+1:

**Review by qmonnet — CHANGES_REQUESTED** [review]
Apologies, I just realised, regarding https://github.com/cilium/cilium/pull/27798: it depends on https://github.com/cilium/cilium/pull/27602, which was not marked for backports on 1.13 (I labelled it now). Is it possible to add it to this PR, please?

**Comment by jibi** [comment]
> Apologies, I just realised, regarding #27798: it depends on #27602, which was not marked for backports on 1.13 (I labelled it now). Is it possible to add it to this PR, please?

@qmonnet there are a few conflicts and I see @julianwiedmann is already taking care of the v1.14 backport, so I'll let him handle also v1.13 (and drop #27798 for now)

**Comment by jibi** [comment]
/test-backport-1.13

**Review by qmonnet — APPROVED** [review]

**Comment by jibi** [comment]
/test-1.24-4.19

**Review by youngnick — APPROVED** [review]

**Pull request merged**